planner/core/planbuilder.go

```diff
@@ -1125,22 +1125,59 @@ func (b *PlanBuilder) buildShow(show *ast.ShowStmt) (Plan, error) {
 		p.Conditions = append(p.Conditions, expr)
 	}
 	if show.Where != nil {
+		var err error
+		var np LogicalPlan
+		np = mockTablePlan
 		conds := splitWhere(show.Where)
 		for _, cond := range conds {
-			expr, _, err := b.rewrite(cond, mockTablePlan, nil, false)
+			var expr expression.Expression
+			expr, np, err = b.rewrite(cond, np, nil, false)
+			if err != nil {
+				return nil, err
+			}
+			if expr != nil {
+				p.Conditions = append(p.Conditions, expr)
```

Comment:
In exectuor builder, `Show` will build selection for the conditions, if the conditions depends on outer plan, I don't think the result will be correct.

Reply:
Yes, this may not resolve the correlated subquery.

Reply:
If the condition is a correlated subquery, the `expr` would be nil here, and it looks like there can be no correlated subquery in the where conditions of `Show`, because the subquery has no outer column to reference?

Reply:
For example, `show columns from t where field < all (select a from t);`?